Terminal.Gui/Views/DatePicker.cs

```diff
@@ -0,0 +1,298 @@
+﻿//
+// DatePicker.cs: DatePicker control
+//
+// Author: Maciej Winnik
+//
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Globalization;
+using System.Linq;
+
+namespace Terminal.Gui.Views;
+
+/// <summary>
+/// The <see cref="DatePicker"/> <see cref="View"/> Date Picker.
+/// </summary>
+public class DatePicker : TextField {
+
+	private DataTable table;
+	private TableView calendar;
+	private DateTime date = DateTime.Now;
+	private List<string> months = new ();
+	private List<string> years = new ();
+
+	/// <summary>
+	/// Format of date. The default is MM/dd/yyyy.
+	/// </summary>
+	public string Format { get; set; } = CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern;
+
+	/// <summary>
+	/// Range of years in the calendar. The default is 1900..2100.
+	/// </summary>
+	public Range YearsRange { get; set; } = 1900..2100;
+
+	/// <summary>
+	/// Get or set the date.
+	/// </summary>
+	public DateTime Date {
+		get => date;
+		set {
+			date = value;
+			Text = date.ToString (Format);
+		}
+	}
+
+	/// <summary>
+	/// Initializes a new instance of <see cref="DatePicker"/>.
+	/// </summary>
+	public DatePicker () : base () => Initialize (date);
+
+	/// <summary>
+	/// Initializes a new instance of <see cref="DatePicker"/> with the specified date.
+	/// </summary>
+	public DatePicker (DateTime date) : base ()
+	{
+		Initialize (date);
+	}
+
+	/// <summary>
+	/// Initializes a new instance of <see cref="DatePicker"/> with the specified date and format.
+	/// </summary>
+	public DatePicker (DateTime date, string format) : base ()
+	{
+		Format = format;
+		Initialize (date);
+	}
+
+	private void Initialize (DateTime date)
```

Comment:
a) No need to use `private`

b) Name this guy `SetInitialProperties`. This is to avoid confusion with `Begin/EndInit`.

Reply:
Done